src-control/Control/ActionRegistry.hs

```diff
@@ -0,0 +1,377 @@
+-- | Registry of monadic actions supporting rollback actions and delayed actions
+-- in the presence of (a-)synchronous exceptions.
```

Comment:
I think we can make more use of the term "commit". It fits a pattern people already know about transactions: committing successfully or if unsuccessful rolling back.

So we can say something like:

> A registry of monadic actions to support working with resources and mutable state in the presences of (a)synchronous exceptions. It works analogously to database transactions: within the \"transaction\" scope we can perform actions (such as resource allocations and state changes) and we can register commit and rollback actions. The commit actions are all executed at the end if the transaction scope is exited successfully, but if an exception is thrown (sync or async) then the rollback actions are executed instead (and the exception is propagated). Commit or rollback actions are executed in the reverse order in which they were registered (which is the natural nesting order when considered as bracketing).